incubator/hnc/pkg/reconcilers/hierarchical_namespace.go

```diff
@@ -30,20 +33,74 @@ type HierarchicalNamespaceReconciler struct {
 }
 
 // Reconcile sets up some basic variables and then calls the business logic.
-// It currently only generates some logs for testing.
+// It currently handles basic creation of namespace and hierarchyconfig.
 func (r *HierarchicalNamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// TODO report error state if the webhook is bypassed - see issue
 	//  https://github.com/kubernetes-sigs/multi-tenancy/issues/459
 	if ex[req.Namespace] {
 		return ctrl.Result{}, nil
 	}
 
+	ctx := context.Background()
 	log := r.Log.WithValues("trigger", req.NamespacedName)
 	log.Info("Reconciling HNS")
 
+	// Names of the self-serve subnamespace and the parent namespace.
+	nm := req.Name
+	pnm := req.Namespace
+
+	// We want to trigger the hierarchyConfig reconciliation on the self-serve subnamespace, since
+	// the parent's 'requiredChildren' field will be deprecated. The HierarchyConfig Reconciler
```

Comment:
This comment won't make much sense in about a week or two - maybe rewrite it as though `requiredChildren` never existed?